transcode.py

```diff
@@ -1,22 +1,47 @@
+from dataclasses import dataclass
 import logging
 import os
 from typing import Optional
+import time
 
 import base_util
 from config import data_base_dir
 
 logger = logging.getLogger(__name__)
 
 
-def try_transcode(input_path, asset_id, extension) -> Optional[str]:
+@dataclass
+class TranscodeOutput:
+    transcoded_file_path: str
+    provenance: dict
+
+
+def try_transcode(input_path, asset_id, extension) -> Optional[TranscodeOutput]:
     logger.info(
         f"Determining if transcode is required for input_path: {input_path} asset_id: ({asset_id}) extension: ({extension})"
     )
+    start_time = time.time()
+
+    provenance = {
+        "activity_name": "Transcoding",
+        "activity_description": "Checks if input needs transcoding, then transcodes if so",
+        "processing_time_ms": 0,
```

Comment:
I would personally prefer something impossible like -1 as a default, so it is clear what the difference is between an actual measurement and a default value.